Library/Homebrew/formula_cellar_checks.rb

```diff
@@ -314,6 +314,24 @@ def check_cpuid_instruction(formula)
     "No `cpuid` instruction detected. #{formula} should not use `ENV.runtime_cpu_detection`."
   end
 
+  def check_binary_arches(formula)
+    return unless formula.prefix.directory?
+    # There is no `binary_executable_or_library_files` method for the generic OS
+    return if !OS.mac? && !OS.linux?
```

Comment:
I'm just curious, why is this the case? I thought the default for this was to use the Linux version and, on macOS, the method was overridden. Is that not what happens with `--generic`?

Reply:
Yes. The default behaviour is to call the method `elf_files`, but that's Linux-only too.